tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -1194,6 +1228,11 @@ void FragmentMetadata::store_v15_or_higher(
 }
 
 void FragmentMetadata::set_num_tiles(uint64_t num_tiles) {
+  if (dense_) {
+    const uint64_t dense_tile_num = tile_num();
+    iassert(num_tiles <= dense_tile_num);
```

Comment:
This is the assert which is tripped without the fix.  When the global order writer doesn't update the fragment domain, each fragment gets the whole domain but a fraction of the tiles.  `tile_num()` is computed using the fragment domain, so they don't agree.